Dockerfile

```diff
@@ -0,0 +1,62 @@
+# syntax=docker/dockerfile:1
+
+# ------------------------------------------------------------------------------
+# Builder stage: build the fractal-engine binary with CGO enabled
+# ------------------------------------------------------------------------------
+FROM golang:bookworm AS builder
+
+# Enable CGO (required for github.com/mattn/go-sqlite3)
+ENV CGO_ENABLED=1 \
+    GO111MODULE=on
+
+WORKDIR /src
+
+# Install build deps for CGO builds (gcc, libc headers, etc.)
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends build-essential ca-certificates && \
+    rm -rf /var/lib/apt/lists/*
+
+# Cache module downloads first
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy the rest of the source
+COPY . .
+
+# Build the application binary
+# Target: cmd/fractal-engine/fractal_engine.go
+RUN --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg/mod \
+    go build -trimpath -ldflags "-s -w" -o /out/fractal-engine ./cmd/fractal-engine
```

Comment:
This doesn't set required LDflags - we should be using nix to build this so it's all set 

https://github.com/dogecoinfoundation/fractal-engine/blob/main/nix/fractalengine.nix#L26-L35